index.bs

```diff
@@ -792,77 +798,84 @@ run these steps:
   1. [=list/iterate|For each=] |node| in |nodes|:
     1. Let |action| be the result of running the [=sanitize a node=] algorithm
-       on |node|.
-    1. Insert |node| and |action| into |m|
+       on |node| with |sanitizer|.
+    1. [=map/Set=] |m|[|node|] to |action|.
   1. [=list/iterate|For each=] |node| in |nodes|:
-    1. If m[node] is `drop`, remove the |node| and all children from |fragment|.
-    1. If m[node] is `block`, replace the |node| with all of its element and text node children from |fragment|.
-    1. If m[node] is `keep`, do nothing.
+    1. If |m|[|node|] is `drop`, [=/remove=] |node|.
+    1. If |m|[|node|] is `block`, [=/replace=] |node| with all of its element and text node children from |fragment|.
+    1. If |m|[|node|] is `keep`, do nothing.
 </div>
 
+Issue(156): The step above needs to explicitly iterate over the children and insert into parent. It could collect them in a variable or do things in place, but this is a bit too imprecise.
+
 <div algorithm="sanitize a node">
-To <dfn>sanitize a node</dfn> named |node| run these steps:
-  1. Let |sanitizer| be the current Sanitizer.
+To <dfn>sanitize a node</dfn> named |node| with |sanitizer| run these steps:
+  1. [=Assert=]: |node| is not a {{Document}} or {{DocumentFragment}} or {{Attr}} or {{DocumentType}} [=/node=].
   1. If |node| is an element node:
-    1. Let |element| be |node|'s element.
+    1. Let |element| be |node|.
     1. [=list/iterate|For each=] |attr| in |element|'s
        [=Element/attribute list=]:
       1. Let |attr action| be the result of running the
          [=sanitize action for an attribute=] algorithm on |attr| and |element|.
-      1. If |attr action| is different from `keep`, remove |attr| from
-        |element|.
+      1. If |attr action| is different from `keep`, [=remove an attribute=] supplying |attr|.
     1. Run the steps to [=handle funky elements=] on |element|.
     1. Let |action| be the result of running the
        [=sanitize action for an element=] on |element|.
     1. Return |action|.
-  1. If |node| is a {{Document}} or {{DocumentFragment}} node and if |node|'s
-     [=parent=] is null: Return `keep`.
   1. If |node| is a {{Comment}} [=node=]:
     1. Let |config| be |sanitizer|'s [=configuration object=], or the
        [=default configuration=] if no [=configuration object=] was given.
     1. If |config|'s [=allow comments option=] [=map/exists=] and `|config|[allowComments]` is `true`: Return `keep`.
     1. Return `drop`.
   1. If |node| is a {{Text}} [=node=]: Return `keep`.
-  1. Return `drop`
+  1. [=Assert=]: |node| is a {{ProcessingInstruction}}
+  1. Return `drop`.
 </div>
 
+Issue(151): The [=sanitize action for an attribute=] algorithm parameters do not match.
+Issue(152): consider renaming "configuration object" to "configuration dictionary".
+Issue(153): consider creating an effective sanitizer config. Also, IDL guarantees that a config is ALWAYS given. The question is really whether the members exists.
```

Comment:
What would be the consequence of this?

Reply:
I haven't fully fleshed it out. Let me make a PR towards that idea in #153 if you don't mind.